evolver/calibration/actions.py

```diff
@@ -86,13 +105,34 @@ def execute(self, state: Dict[str, Any], payload: UserInput) -> Dict[str, Any]:
         if not reference_values or not raw_values:
             raise ValueError(f"Insufficient data to calculate fit for {hardware_name} {vial_key}")
 
-        # Perform the fit calculation (to be implemented based on actual calibration logic)
-        # Perform the fit calculation
-        # TODO: Find out how to call the fit calculation method from the hardware
-        # fit_parameters = self.hardware.calibrate_transformer.calculate_fit(reference_values, raw_values)
-        # TODO: Persist the fit parameters to the Calibrator CalibrationData (see Arik's work) data structure.
-        fit_parameters = [0.5, 0.5]
+        if len(reference_values) != len(raw_values):
+            raise ValueError(
+                f"Reference and raw data lengths do not match for {hardware_name} {vial_key}, "
+                f"Procedure state has {len(reference_values)} reference_values and {len(raw_values)} raw_values recorded so far"
+                f"Unable to calculate fit, you must dispatch the appropriate actions to collect the required data"
+            )
+
+        if not self.hardware.calibrator.Config.output_transformer:
+            raise ValueError(f"No output transformer available for {hardware_name}")
+
+        fit_config = self.hardware.calibrator.Config.output_transformer.fit(reference_values, raw_values)
```

Comment:
actually use the output_transformer on the Temperature Calibrator to calculate the fit.